src/base/misc/disks.c

```diff
@@ -510,7 +510,8 @@ static void image_auto(struct disk *dp)
       leavedos(19);
       return;
     }
-    if (!(set_floppy_chs_by_size(st.st_size, dp) ||
+    if (!(dp->default_cmos == CUSTOM_DISKETTE ||
```

Comment:
Maybe just check above (at line 499)
that chs already set (all values != 0)
and then return immediately?
Looks much simpler than adding another
keyword.

Reply:
I think we do want the `dp->num_secs = ` line that comes later, to calculate the total (CHS-addressable) size though?

Reply:
Sounds reasonable.

Reply:
So would you like to do that?

Reply:
I will get around to it eventually. Still considering how to fix the HD image geometry specifiers too. They're also broken. For example, if we set cylinders X heads Y sectors Z (all specified) then [in disk_reset2](https://github.com/dosemu2/dosemu2/blob/142b4fb9f1a81154b28ed0cd7a144d024177b0d6/src/base/misc/disks.c#L1282) the conditional is not taken, and thus num_secs is never initialised (stays zero) which [makes read_sectors fail](https://github.com/dosemu2/dosemu2/blob/142b4fb9f1a81154b28ed0cd7a144d024177b0d6/src/base/misc/disks.c#L219).